R/module_nested_tabs.R

```diff
@@ -179,16 +208,45 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, reporter) {
       "module { deparse1(modules$label) }."
     )
   )
-
   modules$server_args <- teal.transform::resolve_delayed(modules$server_args, datasets)
-  is_module_server <- isTRUE("id" %in% names(formals(modules$server)))
+  is_module_server <- is_args_used(modules$server, "id")
 
-  args <- c(list(id = id, datasets = datasets), modules$server_args)
-  if (is_reporter_used(modules)) {
+  args <- c(list(id = id), modules$server_args)
+  if (is_args_used(modules$server, "reporter")) {
     args <- c(args, list(reporter = reporter))
   }
 
-  if (is_module_server) {
+  is_datasets_used <- is_args_used(modules$server, "datasets")
+  if (is_datasets_used) {
+    args <- c(args, datasets = datasets)
+  }
```

Comment:
You can probably just do `if(is_args_used(modules$server, "datasets"))` and in other places